libraries/ts-command-line/src/parameters/CommandLineDefinition.ts

```diff
@@ -62,6 +62,11 @@ export interface IBaseCommandLineDefinition {
    *   ordinary String Parameter:  Any value is accepted, including an empty string.
    */
   environmentVariable?: string;
+
+  /**
+   * Custom tab completions for the parameter values
+   */
+  completions?: () => Promise<string[]>;
```

Comment:
Why is this async?

Reply:
I made it async for parameters like `rush build -t` - which can potentially result in a large set of suggestion values.

Reply:
How does being async help though? Would Bash terminate Node.js before it finishes writing the stream?

Reply:
No, but I assumed for some such parameters, consumers might have to make file reads and such to populate the list of values. Now if the consumers have to make async calls, the hook being async would make it easier to implement.

Reply:
Okay, that seems reasonable. 👍